src/Uuid/Uuid.php

```diff
@@ -0,0 +1,274 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Uuid;
+
+/**
+ * @internal
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+final class Uuid
+{
+    const UUID_VARIANT_NCS = 0;
+    const UUID_VARIANT_DCE = 1;
+    const UUID_VARIANT_MICROSOFT = 2;
+    const UUID_VARIANT_OTHER = 3;
+    const UUID_TYPE_DEFAULT = 0;
+    const UUID_TYPE_TIME = 1;
+    const UUID_TYPE_DCE = 4;
+    const UUID_TYPE_NAME = 1;
+    const UUID_TYPE_RANDOM = 4;
+    const UUID_TYPE_NULL = -1;
+    const UUID_TYPE_INVALID = -42;
+
+    public static function uuid_create(int $uuid_type = UUID_TYPE_DEFAULT): string
+    {
+        switch ($uuid_type) {
+            case UUID_TYPE_NAME:
+            case UUID_TYPE_TIME:
+                return self::uuid_generate_time();
+            case UUID_TYPE_DCE:
+            case UUID_TYPE_RANDOM:
+            case UUID_TYPE_DEFAULT:
+                return self::uuid_generate_random();
+            default:
+                trigger_error("Unknown/invalid UUID type '%d' requested, using default type instead", E_USER_WARNING);
+                return self::uuid_generate_random();
+        }
+    }
+
+    public static function uuid_is_valid(string $uuid): bool
+    {
+        return null !== self::uuid_parse_as_array($uuid);
+    }
+
+    public static function uuid_compare(string $uuid1, string $uuid2)
+    {
+        if (null === self::uuid_parse_as_array($uuid1)) {
+            return false;
+        }
+
+        if (null === self::uuid_parse_as_array($uuid2)) {
+            return false;
+        }
+
+        return $uuid1 <=> $uuid2;
+    }
+
+    public static function uuid_is_null(string $uuid): bool
+    {
+        return '00000000-0000-0000-0000-000000000000' === $uuid;
+    }
+
+    public static function uuid_type(string $uuid)
+    {
+        if (null === $parsed = self::uuid_parse_as_array($uuid)) {
+            return false;
+        }
+
+        if (self::uuid_is_null($uuid)) {
+            return self::UUID_TYPE_NULL;
+        }
+
+        return ($parsed['time_hi_and_version'] >> 12) & 0xF;
+    }
+
+    public static function uuid_variant(string $uuid)
+    {
+        if (null === $parsed = self::uuid_parse_as_array($uuid)) {
+            return false;
+        }
+
+        if (self::uuid_is_null($uuid)) {
+            return self::UUID_TYPE_NULL;
+        }
+
+        if (($parsed['clock_seq'] & 0x8000) == 0) {
+            return self::UUID_VARIANT_NCS;
+        }
+        if (($parsed['clock_seq'] & 0x4000) == 0) {
+            return self::UUID_VARIANT_DCE;
+        }
+        if (($parsed['clock_seq'] & 0x2000) == 0) {
+            return self::UUID_VARIANT_MICROSOFT;
+        }
+
+        return self::UUID_VARIANT_OTHER;
+    }
+
+    public static function uuid_time(string $uuid)
+    {
+        if (PHP_INT_SIZE !== 8) {
```

Comment:
should be fully-qualified to be resolved at compile time and let OPCache optimize dead code due to this condition (same reason that for our PHP version checks)

Reply:
Fixed. Thanks